chemicalx/models/deepdrug.py

```diff
@@ -1,14 +1,102 @@
 """An implementation of the DeepDrug model."""
 
-from .base import UnimplementedModel
+from typing import Optional
+
+import torch
+from torchdrug.data import PackedGraph
+from torchdrug.layers import GraphConv, MaxReadout
+
+from chemicalx.constants import TORCHDRUG_NODE_FEATURES
+from chemicalx.data import DrugPairBatch
+from chemicalx.models import Model
 
 __all__ = [
     "DeepDrug",
 ]
 
 
-class DeepDrug(UnimplementedModel):
+class DeepDrug(Model):
     """An implementation of the DeepDrug model.
 
-    .. seealso:: https://github.com/AstraZeneca/chemicalx/issues/14
+    .. based on:: https://www.biorxiv.org/content/biorxiv/early/2020/11/10/2020.11.09.375626.full.pdf
     """
+
+    def __init__(
+        self,
+        *,
+        context_channels: int = 112,
+        molecule_channels: int = TORCHDRUG_NODE_FEATURES,
+        num_gcn_layers: int = 4,
+        gcn_layer_hidden_size: int = 64,
+        out_channels: int = 1,
+        dropout_rate: float = 0.1,
+    ):
+        """Instantiate the DeepDrug model.
+
+        :param context_channels: The number of contexts used for learning. Set to 0 if not using any context. In that
+        case make sure you're not calling forward() with the context feature tensor.
+        :param molecule_channels: The number of molecular features.
+        :param num_gcn_layers: Number of GCN layers.
+        :param gcn_layer_hidden_size: number of hidden units in GCN layers
+        :param out_channels: The number of output channels.
+        :param dropout_rate: Dropout rate on the final fully-connected layer.
+        """
+        super(DeepDrug, self).__init__()
+        self.num_gcn_layers = num_gcn_layers
+        self.gcn_layer_hidden_size = gcn_layer_hidden_size
+        self.graph_convolution_first = GraphConv(molecule_channels, self.gcn_layer_hidden_size, batch_norm=True)
+
+        # add remaining GCN layers
+        self.gcn_layer_list = torch.nn.ModuleList()
+        for i in range(num_gcn_layers - 1):
```

Comment:
`i` is not used in this loop

```suggestion
         for _ in range(num_gcn_layers - 1):
```